### PR TITLE
Fix `link-action` redirect network error

### PR DESCRIPTION
**Pull request by nanguanlin6**
image

### CODE REVIEW

**Comment by lunny** [comment]
We need a `return` on line 171

**Review by 6543 — APPROVED** [review]

File: routers/web/repo/projects.go [hunk 1]
```diff
@@ -161,14 +161,15 @@ func NewProjectPost(ctx *context.Context) {
...
 func ChangeProjectStatus(ctx *context.Context) {
-	toClose := false
+	var toClose bool
 	switch ctx.Params(":action") {
 	case "open":
 		toClose = false
 	case "close":
 		toClose = true
 	default:
 		ctx.Redirect(ctx.Repo.RepoLink + "/projects")
```
Comment (6543) [inline routers/web/repo/projects.go]: Should nt be this also changed @lng2020 ?

Comment (nanguanlin6) [reply routers/web/repo/projects.go]: Reasonable. Changed.

**Review by 6543 — APPROVED** [review]

**Comment by 6543** [comment]
@lng2020 since what version do this issue exist?

**Comment by lng2020** [comment]
From my test, 1.20 is okay. Probably because they changed the frontend redirect logic code several months ago.

**Comment by yp05327** [comment]
The behavior of `link-action` was changed in that PR.

It seems that all of them are broken.
For example:
open of close a milestone also has this issue.
image

**Comment by lng2020** [comment]
Yes. If the frontend uses `link-action`, the backend counterpart should use `JSONRedirect` instead of `Redirect`.

**Comment by lng2020** [comment]
From my test, the milestone and project are the only two occasions that encounter this error. Will fix the milestone issue in this PR too.

**Review by delvh — APPROVED** [review]

**Review by puni9869 — APPROVED** [review]

**Pull request merged**